### PR TITLE
Fix crash on changing resource type of dictionary new key/value while editing the resource.

### LINKED ISSUE #88286: Changing Object Class of new Key/Value form dictionary argument while the Object is opened will crash Godot
### Tested versions

reproducible in : v4.3.dev.custom_build [71a6ee64a], 4.2.1

### System information

macOS 13.3.0 - Vulkan (Forward+) - integrated Apple M1 Pro - Apple M1 Pro (8 Threads)

### Issue description

When you change the class of the new value/key of dictionary editor while Object Editor is opened Godot will crash.

https://github.com/godotengine/godot/assets/66184050/e292092b-fa22-4909-bb9b-219d8cb37ff8

### Steps to reproduce

1. create a script that export a dict 
2. on the new key/value change type to Object
3. choose a resource
4. open the editor so that an Inspector is shown
5. change the object to another class of resource 
6. Godot crashes

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by ajreckof**
Fixes #88286

### CODE REVIEW

File: editor/editor_resource_picker.cpp [hunk 1]
```diff
@@ -467,8 +467,9 @@ void EditorResourcePicker::_edit_menu_cbk(int p_which) {
 
 			EditorNode::get_editor_data().instantiate_object_properties(obj);
 
+			Ref<Resource> old_edited_resource = edited_resource; // prevent freeing of the object until the end of the update of the resource see #88286
```
Comment (akien-mga) [inline editor/editor_resource_picker.cpp:470]:
```suggestion
			// Prevent freeing of the object until the end of the update of the resource (GH-88286).
			Ref<Resource> old_edited_resource = edited_resource;
```

File: editor/editor_resource_picker.cpp [hunk 2]
```diff
@@ -467,8 +467,10 @@ void EditorResourcePicker::_edit_menu_cbk(int p_which) {
 
 			EditorNode::get_editor_data().instantiate_object_properties(obj);
 
+			// Prevent freeing of the object until the end of the update of the resource (GH-88286).
+			Ref<Resource> old_edited_resource = edited_resource;
 			edited_resource = Ref<Resource>(resp);
-			emit_signal(SNAME("resource_changed"), edited_resource);
+			emit_signal(SNAME("resource_changed"), Ref<Resource>(resp));
```
Comment (KoBeWi) [inline editor/editor_resource_picker.cpp:473]: This part seems unnecessary.

Comment (ajreckof) [reply editor/editor_resource_picker.cpp:473]: Thanks this was needed for a previous idea on how to fix this and forgot to remove the change will fix it when home.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**